### PR TITLE
Various improvements

### PR DESCRIPTION
**Pull request by Tom94**

### CODE REVIEW

File: osu.Framework/Graphics/UserInterface/TextBox.cs [hunk 1]
```diff
@@ -55,42 +56,56 @@ public class TextBox : Container
...
-        private void load(BaseGame game)
+        public TextBox()
         {
-            this.game = game;
-
             Masking = true;
             CornerRadius = 3;
 
-            Add(background = new Box
+            Add(new Drawable[]
```
Comment (UselessToucan) [inline osu.Framework/Graphics/UserInterface/TextBox.cs]: I guess it should be `Children = new Drawable[]{...}`.

File: osu.Framework/Graphics/UserInterface/TextBox.cs [hunk 2]
```diff
@@ -683,26 +698,11 @@ protected override bool OnFocus(InputState state)
 
         private void unbindInput()
         {
-            textInput?.Deactivate(this);
+            textInput.Deactivate(this);
```
Comment (UselessToucan) [inline osu.Framework/Graphics/UserInterface/TextBox.cs]: Why did you change this?

File: osu.Framework/Graphics/UserInterface/TextBox.cs (hunk 1 above)
Comment (Tom94) [inline osu.Framework/Graphics/UserInterface/TextBox.cs]: Both are fine, whereas ` Children =` overwrites existing children and `Add` just adds them. We haven't yet decided for a preferred default when no children are present yet, so I'd leave such a refactor for later.
